src/Sulu/Bundle/MediaBundle/Resources/js/containers/Form/fields/ImageMap.js

```diff
@@ -0,0 +1,154 @@
+// @flow
+import React from 'react';
+import {observable, toJS} from 'mobx';
+import debounce from 'debounce';
+import userStore from 'sulu-admin-bundle/stores/userStore';
+import type {FieldTypeProps} from 'sulu-admin-bundle/types';
+import type {BlockError} from 'sulu-admin-bundle/containers/Form/types';
+import FieldRenderer from 'sulu-admin-bundle/containers/FieldBlocks/FieldRenderer';
+import jsonpointer from 'json-pointer';
+import ImageMapContainer from '../../ImageMap';
+import type {Value as ImageMapValue, RenderHotspotFormCallback} from '../../ImageMap/types';
+
+const MISSING_TYPE_ERROR_MESSAGE = 'The "image_map" field type needs at least one type to be configured!';
+
+export default class ImageMap extends React.Component<FieldTypeProps<ImageMapValue>> {
+    handleFinish = debounce(() => {
```

Comment:
We don't have a `mouse up` event here

Reply:
Yeah, but couldn't the component having that mouse up event only trigger its `onChange` call once it is finished? Then calling components would not have to care about this. Or am I missing something?

Reply:
Not without a lot of refactoring, i guess, because now the state lives in the calling component, not in the `ModifiableCircle` or `ModifiableRectangle`

Reply:
And I don't think that using a debouncer makes that a lot more complicated

Reply:
The problem is less that it is terribly complicated, but more that the caller might forget about this. Gonna have a closer look at that later.

Reply:
We've decided that the `RectangleSelection` and `CircleSelection` have an `onFinish` callback, and this callback calls this `handleFinish` function. That makes the `debounce` unnecessary.

Reply:
fixed